src/main/java/org/elasticsearch/river/kafka/KafkaWorker.java

```diff
@@ -100,22 +97,10 @@ private void consumeMessagesAndAddToBulkProcessor(final KafkaStream stream) {
                 final MessageAndMetadata messageAndMetadata = consumerIterator.next();
                 logMessage(messageAndMetadata);
 
-                messageSet.add(messageAndMetadata);
```

Comment:
@mariamhakobyan we can agree to disagree -- but, it seems that it's just extra load on the river, and thus on the Es machine that's hosting it...

1. That BulkProcessor bug you mentioned has a fixed a long time ago.
2. If Elasticsearch is stopped, so is the river. So using a Set<> doesn't add any fault tolerance here.
3. BulkProcessor.add() appends to a BulkRequest, which is in-fact a List<>. So even if the flush interval has not reached, it won't matter.
4. Pending bulk requests / concurrency issues are handled by the BulkProcessor as well, so there's really no need for that extra Set.
5. Fault tolerance should really by handled by the bulk processor handlers, by acking or failing Kafka messages.